src/entrypoint.sh

```diff
@@ -60,9 +66,17 @@ function git_init() {
   git config --global --add safe.directory /github/workspace
   git lfs install
 
-  git config --global "credential.https://${SOURCE_REPO_HOSTNAME}.helper" "!gh auth git-credential"
-  gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${GITHUB_TOKEN}"
-
+  if [[ -n "${SOURCE_REPO_GITHUB_TOKEN}" ]]; then
+      if [[ -n "${GITHUB_TOKEN}" ]]; then
+        export GITHUB_TOKEN_BK="${GITHUB_TOKEN}"
+        unset GITHUB_TOKEN
```

Comment:
this part is needed because if GITHUB_TOKEN is set, gh cli will use it as default value

Reply:
general question: why not using `if [[ -z "${SSH_PRIVATE_KEY_SRC}" ]] &>/dev/null` and don't set the "incompatible" configs instead of unsetting the default GITHUB_TOKEN var?

Reply:
like this 

```sh
function git_init() {
  echo "::group::git init"
  info "set git global configuration"

  git config --global user.email "github-action@actions-template-sync.noreply.${SOURCE_REPO_HOSTNAME}"
  git config --global user.name "${GITHUB_ACTOR}"
  git config --global pull.rebase false
  git config --global --add safe.directory /github/workspace
  git lfs install

  if [[ -z "${SSH_PRIVATE_KEY_SRC}" ]] &>/dev/null; then
    gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_REPO_GITHUB_TOKEN}"
    gh auth setup-git --hostname "${SOURCE_REPO_HOSTNAME}"

    gh auth status
  fi

  echo "::endgroup::"
}

Should that not also solve ihe ssh issue?
```

Reply:
The issue is not only related to `ssh`. The issue is if the `GITHUB_TOKEN` variable is set, `gh` cli will use the token per default and throw an error.
Imagine you have 2 differrent tokens `GITHUB_TOKEN` and  `SOURCE_REPO_GITHUB_TOKEN`, then this code path will throw an error.

Reply:
hmm as an idea to be able to support your use case without the more generic one (KISS).
Let's use your idea and remove the `SOURCE_REPO_GITHUB_TOKEN` again (I will add this in future).
With that in mind, lets use your idea and put the auth path in the else of the ssh check
![Screenshot 2023-01-02 at 16 33 55](https://user-images.githubusercontent.com/8027933/210252253-626be61b-661a-42cd-8318-d028abad6b5c.png)

What do you think? I can do the changes or you can do them. As you like.

Sorry for the long discussions..